pkg/controller/pipe/integration.go

```diff
@@ -210,6 +218,27 @@ func CreateIntegrationFor(ctx context.Context, c client.Client, binding *v1.Pipe
 	return &it, nil
 }
 
+// shouldHandleNoErrorHandler determines the runtime version and checks on noErrorHandler that is configured for this version.
+func shouldHandleNoErrorHandler(ctx context.Context, c client.Client, it *v1.Integration) bool {
+	var runtimeVersion string
+	if it.Spec.Traits.Camel != nil && it.Spec.Traits.Camel.RuntimeVersion != "" {
+		runtimeVersion = it.Spec.Traits.Camel.RuntimeVersion
+	} else if pl, err := platform.GetForResource(ctx, c, it); err == nil && pl != nil {
```

Comment:
No need to peek the platform. At trait level stage, we must have already set the Integration.Status.RuntimeVersion.

Reply:
No, I don't think so as the integration resource is being created by the pipe controller at this stage. it has no status yet as it has not been created yet

Reply:
Not here. It will be available in the trait. Even more, this has to be executed during the Integration.Status.Phase runtime only.

Reply:
yes moved code to trait, thanks!